packages/backend-data/src/factory.ts

```diff
@@ -149,7 +156,38 @@ class DataGenerator implements ConstructContainerEntryGenerator {
         ? this.props.schema.schemas
         : [this.props.schema];
 
-      schemas.forEach((schema) => {
+      // get the branch name and use the imported table map for that key
+      // use the sandbox key when the branch name is not available (e.g. in the sandbox deployment)
+      const amplifyBranchName =
+        scope.node.tryGetContext('amplifyEnvironmentName') ?? 'sandbox';
```

Comment:
could we use these?

https://github.com/aws-amplify/amplify-backend/blob/main/packages/platform-core/src/cdk_context_key.ts

Some examples:
https://github.com/aws-amplify/amplify-backend/blob/a6ef0f3afe87be8edf9c35d5352a35ada423c810/packages/backend-data/src/factory.ts#L245

https://github.com/aws-amplify/amplify-backend/blob/a6ef0f3afe87be8edf9c35d5352a35ada423c810/packages/backend/src/default_stack_factory.test.ts#L11-L13